Makefile

```diff
@@ -88,7 +88,11 @@ GOOS ?= $(shell go env GOOS)
 GOARCH ?= $(shell go env GOARCH)
 GOPATH ?= $(shell go env GOPATH)
 BUILD_DIR ?= ./out
-$(shell mkdir -p $(BUILD_DIR))
+# Re-attach case-sensitive sparseimage in case it got detached by e.g. a reboot
+ifeq ($(GOOS),darwin)
+	IGNORED := $(shell test ! -e $(BUILD_DIR) -a -f $(BUILD_DIR).sparseimage && hdiutil attach -mountpoint $(BUILD_DIR) $(BUILD_DIR).sparseimage)
```

Comment:
You probably want to leave "out" as-is, and only do "out/buildroot" as a sparse image ?

Reply:
> I think the other targets are fine (with out), so this should be more buildroot-specific ?

Yes, they seem to be, which is why the check for the case-sensitive filesystem is only in the `minikube_iso` target. So you don't need to run `make case-sensitive-build-dir` unless you want to build that target.

But when you switch to a sparseimage, I don't know why it should be a problem switching all of `./out`. It should be completely transparent. Doing it just for the buildroot dir makes the patch a bit more complex for no benefit afaict. Let me know if I'm missing something, and please confirm that this change really is required!

BTW, I noticed that support for changing `$(BUILD_DIR)` is pretty incomplete: there are plenty of verbatim `out/` strings in targets, dependencies and commands, so changing it from the default value doesn't seem to work in general. Is this something you want to see changed?

Reply:
> Doing it just for the buildroot dir makes the patch a bit more complex for no benefit afaict.

I realized that it doesn't actually make it significantly more complex, so I have changed the PR accordingly.

Please let me know if you want me to squash the commits in the PR? I've already rebased on the latest origin/master to hopefully get rid of the functional test failures.